### PR TITLE
Add .NET 9 target

### PR DESCRIPTION
**Pull request by mus65**
I had to disable a SonarSource analyzer and made use of the new `Lock` type, see separate Commits.

CI is expected to fail because of missing .NET 9 binaries in AppVeyor. Build and tests pass locally.

Can be merged when .NET 9 final is released and AppVeyor supports it.

### CODE REVIEW

**Comment by Rob-Hague** [comment]
Can we have a basic Lock polyfill to save on the #ifdefs? e.g.

```c#
#if !NET9_0_OR_GREATER

namespace Renci.SshNet.Common;

internal sealed class Lock
{
    public bool TryEnter() => Monitor.TryEnter(this);
    public void Exit() => Monitor.Exit(this);
}

```

**Comment by mus65** [comment]
Makes sense, pushed.

**Comment by drieseng** [comment]
@mus65, you'll also want to update **global.json**.

File: .editorconfig
Comment (drieseng) [inline .editorconfig:199]: Where was this (falsely) reported?

File: appveyor.yml
Comment (drieseng) [inline appveyor.yml]: Do we want to switch from .NET 8.0 (LTS) to .NET 9.0 (STS, currently still in preview)?
Does AppVeyor even support .NET 9.0?

File: test/Renci.SshNet.AotCompatibilityTestApp/Renci.SshNet.AotCompatibilityTestApp.csproj
Comment (drieseng) [inline test/Renci.SshNet.AotCompatibilityTestApp/Renci.SshNet.AotCompatibilityTestApp.csproj]: Do we want this?

File: test/Renci.SshNet.Benchmarks/Renci.SshNet.Benchmarks.csproj
Comment (drieseng) [inline test/Renci.SshNet.Benchmarks/Renci.SshNet.Benchmarks.csproj:5]: Do we want this?

File: test/Renci.SshNet.IntegrationBenchmarks/Renci.SshNet.IntegrationBenchmarks.csproj
Comment (drieseng) [inline test/Renci.SshNet.IntegrationBenchmarks/Renci.SshNet.IntegrationBenchmarks.csproj:5]: Do we want this?

File: test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
Comment (drieseng) [inline test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj]: Do we want this?

**Comment by drieseng** [comment]
@Rob-Hague, are planning on merging this before the release of .NET 9.0? In theory, there may even be API changes - I'm thinking of the **Lock** class - between now and the release. It's highly unlikely though.

File: .editorconfig
Comment (mus65) [reply .editorconfig:199]: I mentioned this in the commit (should have written it in the PR in hindsight). This causes false positives with `Debug.Assert` when a custom message is supplied. This is only a problem since .NET 9 because of this change: https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#debugassert-now-reports-assert-condition-by-default

File: appveyor.yml
Comment (mus65) [reply appveyor.yml]: > Do we want to switch from .NET 8.0 (LTS) to .NET 9.0 (STS, currently still in preview)?

Imho it makes sense to test against the latest .NET to catch any regressions. Would even be better to test against all of them, but that's probably not worth the CI overhead. That's up to you.

I didn't want this to be merged while .NET 9 is still in preview, that's why it's still draft.

> Does AppVeyor even support .NET 9.0?

it doesn't. I'm just hoping that it supports it when .NET 9 final is released. If not, the SDK could probably be installed manually somehow.

**Comment by mus65** [comment]
@drieseng @Rob-Hague I updated global.json . I had to reference RC1 explicitly though, so I have to change it when final is released.

I left it a draft because it's still preview and wouldn't expect it to be merged before .NET 9 final. RC1 is already released, so I wouldn't expect any more API changes.

**Comment by Rob-Hague** [comment]
Yeah, we will merge once .NET 9 is released

Newer targets can be used on appveyor with apt-get, as in the opposite of #1352

File: test/Renci.SshNet.AotCompatibilityTestApp/Renci.SshNet.AotCompatibilityTestApp.csproj
Comment (Rob-Hague) [reply test/Renci.SshNet.AotCompatibilityTestApp/Renci.SshNet.AotCompatibilityTestApp.csproj]: Fine either way

File: test/Renci.SshNet.Benchmarks/Renci.SshNet.Benchmarks.csproj
Comment (Rob-Hague) [reply test/Renci.SshNet.Benchmarks/Renci.SshNet.Benchmarks.csproj:5]: Yes

File: test/Renci.SshNet.IntegrationBenchmarks/Renci.SshNet.IntegrationBenchmarks.csproj
Comment (Rob-Hague) [reply test/Renci.SshNet.IntegrationBenchmarks/Renci.SshNet.IntegrationBenchmarks.csproj:5]: Yes

File: test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
Comment (Rob-Hague) [reply test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj]: Let's keep a net8.0 target just for local testing

```suggestion
    <TargetFrameworks>net48;net8.0;net9.0</TargetFrameworks>
```

**Comment by mus65** [comment]
This is now only blocked by AppVeyor not supporting .NET 9. apt-get is not going work either because the dotnet9 package is not available in Ubuntu 22.04 (see https://launchpad.net/ubuntu/+source/dotnet9 ).

@Rob-Hague has there been any thought about moving to GitHub actions?

**Comment by Rob-Hague** [comment]
I seem to remember @WojciechNagorski trying it but I can't find a PR so may be imagining

**Comment by Rob-Hague** [comment]
Feel free to give a go though

**Review by Rob-Hague — APPROVED** [review]

**Pull request merged**
